clang/test/CIR/CodeGen/attribute-annotate-multiple.cpp

```diff
@@ -0,0 +1,169 @@
+// BEFORE-SAME: unit = <"[[PATH]]attribute-annotate-multiple.cpp" : 
+// BEFORE-SAME: !cir.array<!s8i x [[FILENAME_LEN]]>>, 
+// BEFORE-SAME: lineno = <12>, args = #cir.const_struct<{#cir.global_view<@".str1"> :
+// BEFORE-SAME: !cir.ptr<!cir.array<!s8i x 3>>, #cir.int<22> : !s32i}>
+// BEFORE:  cir.global "private"  constant internal dsolocal @".str1"
+// BEFORE-SAME: #cir.const_array<"os\00" : !cir.array<!s8i x 3>> :
+// BEFORE-SAME: !cir.array<!s8i x 3> {alignment = 1 : i64}
+
+// AFTER: ![[ANNO_STRUCT_T0:.*]] = !cir.struct<struct  {!cir.ptr<!cir.void>,
+// AFTER-SAME: !cir.ptr<!cir.void>, !cir.ptr<!cir.void>, !cir.int<u, 32>, !cir.ptr<!cir.void>}>
+// AFTER: ![[ANNO_STRUCT_T1:.*]] = !cir.struct<struct {!cir.ptr<!cir.array<!cir.int<s, 8> x 3>>, !cir.int<s, 32>}>
+// AFTER: cir.global "private"  constant cir_private dsolocal @".str.0.llvm.metadata" = 
+// AFTER-SAME: #cir.const_array<"withargs" : !cir.array<!s8i x 8>> :
+// AFTER-SAME: !cir.array<!s8i x 8> {section = "llvm.metadata"}
+// AFTER: cir.global "private" constant cir_private dsolocal @".str.1.llvm.metadata" =
+// AFTER-SAME: #cir.const_array<"[[PATH:.*]]attribute-annotate-multiple.cpp" : 
+// AFTER-SAME: !cir.array<!s8i x [[FILENAME_LEN:[0-9]+]]>> : 
+// AFTER-SAME: !cir.array<!s8i x [[FILENAME_LEN]]> {section = "llvm.metadata"}
+// AFTER: cir.global "private"  constant cir_private dsolocal @".args.0.llvm.metadata" =
+// AFTER-SAME: #cir.const_struct<{#cir.global_view<@".str"> :
+// AFTER-SAME: !cir.ptr<!cir.array<!s8i x 3>>, #cir.int<12> : !s32i}> :
+// AFTER-SAME: ![[ANNO_STRUCT_T1]] {section = "llvm.metadata"}
+// AFTER: cir.global "private"  constant cir_private dsolocal @".str.2.llvm.metadata" =
+// AFTER-SAME: #cir.const_array<"noargvar" : !cir.array<!s8i x 8>> :
+// AFTER-SAME: !cir.array<!s8i x 8> {section = "llvm.metadata"}
+// AFTER: cir.global "private"  constant cir_private dsolocal @".str.3.llvm.metadata" = 
+// AFTER-SAME: #cir.const_array<"noargfunc" : !cir.array<!s8i x 9>> :
+// AFTER-SAME: !cir.array<!s8i x 9> {section = "llvm.metadata"} 
+// AFTER: cir.global "private"  constant cir_private dsolocal @".str.4.llvm.metadata" = 
+// AFTER-SAME: #cir.const_array<"withargfunc" : !cir.array<!s8i x 11>> :
+// AFTER-SAME: !cir.array<!s8i x 11> {section = "llvm.metadata"}
+// AFTER: cir.global "private"  constant cir_private dsolocal @".args.1.llvm.metadata" =
+// AFTER-SAME: #cir.const_struct<{#cir.global_view<@".str1"> :
+// AFTER-SAME: !cir.ptr<!cir.array<!s8i x 3>>, #cir.int<23> : !s32i}> :
+// AFTER-SAME: ![[ANNO_STRUCT_T1]] {section = "llvm.metadata"}
+// AFTER: cir.global "private"  constant cir_private dsolocal @".args.2.llvm.metadata" =
+// AFTER-SAME: #cir.const_struct<{#cir.global_view<@".str1"> :
+// AFTER-SAME: !cir.ptr<!cir.array<!s8i x 3>>, #cir.int<22> : !s32i}> :
+// AFTER-SAME: ![[ANNO_STRUCT_T1]] {section = "llvm.metadata"}
+
+// AFTER: cir.global  appending @llvm.global.annotations =
```

Comment:
The only thing lowering prepare should do is to create a global `@cir.global.annotations`, but there shouldn't be more lowering done here. Please also don't use cir.const_array for it either. This is an example of how it should look like:
```
cir.global appending @llvm.global.annotations = #cir.global_annotation<(#cir.annotation<A>, #cir.annotation<B>, ...)>
```
Please add an alias for `#cir.annotation` so it can be outlined in the output. LLVM lowering should then lower `#cir.global_annotation` to all this LLVM specific new string globals and pointers to them.